File: src/main/java/org/droidmate/analyzer/exploration/IExplorationResult.java
package org.droidmate.analyzer.exploration;

import org.droidmate.analyzer.api.IApi;

import java.nio.file.Path;
import java.util.List;

/**
 * Processed exploration results
 */
public interface IExplorationResult {
    Path getExplDir();

    boolean hasCrashed();

    List<IApi> getApiList();

    double getSize();

    String toBrackedNotation();

    int getNrWidgetsExplored();

    int getNrWidgetsObserved();
}

File: src/main/kotlin/org/droidmate/analyzer/exploration/IExplorationResult.kt
package org.droidmate.analyzer.exploration

import org.droidmate.analyzer.api.IApi

import java.nio.file.Path
import java.time.Duration

/**
 * Processed exploration results
 */
interface IExplorationResult {
    val explDir: Path

    fun hasCrashed(): Boolean

    val apiList: List<IApi>

    val size: Double

    fun toSortedBracedNotation(): String

    fun toBracedNotation(): String

    fun getPolicies() : List<IApi>

    val duration: Duration

    val nrWidgetsExplored: Int

    val nrWidgetsObserved: Int
}
